lib/network/connection_manager.go

```diff
@@ -45,6 +46,14 @@ func NewConnectionManager(
 	}
 }
 
+func (c *ConnectionManager) SetBroadcaster(broadcaster Broadcaster) {
+	c.broadcaster = broadcaster
+}
+
+type Broadcaster interface {
```

Comment:
Why do we need this interface ?

Reply:
It is used for both `TestBroadcaster` and `SimpleBroadcaster`.